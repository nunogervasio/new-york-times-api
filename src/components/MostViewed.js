import React, { Component } from "react";
import MostViewedCard from "./mostViewedCard";
import { Header } from "semantic-ui-react";
import { sliceArticals } from "../utils";
import { format } from "date-fns";
import { APIKEY } from "../config";

class MostViewed extends Component {
  state = {
    data: []
  };

  componentDidMount() {
    fetch(
      `https://api.nytimes.com/svc/mostpopular/v2/viewed/1.json?api-key=${APIKEY}`
    )
      .then(responce => responce.json())
      .then(responceData =>
        this.setState({
          data: responceData.results
        })
      )
      .catch(error => {
        console.log("Error fetching and parsing data...", error);
      });
  }

  render() {
    return (
      <div>
        <div className="section-header">
          <Header size="large" textAlign="center">
            Most Viwed Articals Today
          </Header>
        </div>
        {sliceArticals(this.state.data, 3).map(obj => (
          <MostViewedCard
            key={obj.id}
            artical_url={obj.url}
            headline={obj.title}
            summary={obj.abstract}
            pub_date={format(obj.published_date, "MMM D")}
            img_url={obj.media[0]["media-metadata"][2].url}
            section={obj.section}
          />
        ))}
      </div>
    );
  }
}

export default MostViewed;
