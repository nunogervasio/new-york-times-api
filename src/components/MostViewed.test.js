import React from "react";
import { shallow } from "enzyme";
import MostViewed from "./MostViewed";

describe("MostViewed", () => {
  const mostViewed = shallow(<MostViewed />);

  it("renders properly", () => {
    expect(mostViewed).toMatchSnapshot();
  });
  it("renders MostViewedCard", () => {});
});
