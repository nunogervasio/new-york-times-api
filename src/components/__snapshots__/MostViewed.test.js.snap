// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MostViewed renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MostViewed />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="section-header"
        >
          <Header
            size="large"
            textAlign="center"
          >
            Most Viwed Articals Today
          </Header>
        </div>,
        Array [],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Header
            size="large"
            textAlign="center"
          >
            Most Viwed Articals Today
          </Header>,
          "className": "section-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Most Viwed Articals Today",
            "size": "large",
            "textAlign": "center",
          },
          "ref": null,
          "rendered": "Most Viwed Articals Today",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="section-header"
          >
            <Header
              size="large"
              textAlign="center"
            >
              Most Viwed Articals Today
            </Header>
          </div>,
          Array [],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Header
              size="large"
              textAlign="center"
            >
              Most Viwed Articals Today
            </Header>,
            "className": "section-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Most Viwed Articals Today",
              "size": "large",
              "textAlign": "center",
            },
            "ref": null,
            "rendered": "Most Viwed Articals Today",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": true,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
